olp-cpp-sdk-dataservice-read/tests/PartitionsCacheRepositoryTest.cpp

```diff
@@ -132,4 +136,42 @@ TEST(PartitionsCacheRepositoryTest, DefaultExpiry) {
   }
 }
 
+TEST(PartitionsCacheRepositoryTest, QuadTree) {
+  const auto hrn = HRN::FromString(kCatalog);
+  const auto layer = "layer";
+  const auto version = 0;
+  const auto tile_key = olp::geo::TileKey::FromHereTile("23618364");
+  const auto depth = 2;
+
+  {
+    SCOPED_TRACE("Put/Get quad tree");
+
+    auto stream = std::stringstream(kQuadkeyResponse);
+    read::QuadTreeIndex quad_tree(tile_key, depth, stream);
+    std::shared_ptr<KeyValueCache> cache =
+        olp::client::OlpClientSettingsFactory::CreateDefaultCache({});
+    repository::PartitionsCacheRepository repository(hrn, cache);
+
+    repository.Put(layer, tile_key, depth, quad_tree, version);
+    const auto result = repository.Get(layer, tile_key, depth, version);
+
+    ASSERT_FALSE(result.IsNull());
+    ASSERT_EQ(*result.GetRawData(), *quad_tree.GetRawData());
+  }
+
+  {
+    SCOPED_TRACE("Empty quad tree");
+
+    read::QuadTreeIndex quad_tree;
+    std::shared_ptr<KeyValueCache> cache =
+        olp::client::OlpClientSettingsFactory::CreateDefaultCache({});
+    repository::PartitionsCacheRepository repository(hrn, cache);
+
+    repository.Put(layer, tile_key, depth, quad_tree, version);
+    const auto result = repository.Get(layer, tile_key, depth, version);
+
+    ASSERT_TRUE(result.IsNull());
+  }
```

Comment:
Usually, unit tests are unit testing the class only, which means it should work with mocked interfaces, e.g. mocked KeyValueCache, and test that it places the correct blob under the correct key.

Reply:
DefaultCache replaced by MockCache :-)
Not sure about generating cache key in the test.
Currently, key value captured during Put() call and checked when trying to Get() QuadTree.

Reply:
Yes, it was just a loud thought. I think it looks good for now.